bundles/org.openhab.binding.froniuswattpilot/src/main/java/org/openhab/binding/froniuswattpilot/internal/FroniusWattpilotHandler.java

```diff
@@ -0,0 +1,308 @@
+                            }
+                            ampere = qt.intValue();
+                        }
+                        case DecimalType dt -> ampere = dt.intValue();
+                        default -> {
+                            logger.debug("Command has wrong type, QuantityType or DecimalType required!");
+                            return;
+                        }
+                    }
+                    client.sendCommand(new SetChargingCurrentCommand(ampere));
+                    break;
+                case CHANNEL_PV_SURPLUS_THRESHOLD:
+                    int watts;
+                    switch (command) {
+                        case QuantityType<?> qt -> {
+                            qt = qt.toUnit(Units.WATT);
+                            if (qt == null) {
+                                logger.debug("Failed to convert QuantityType to WATT!");
+                                return;
+                            }
+                            watts = qt.intValue();
+                        }
+                        case DecimalType dt -> watts = dt.intValue();
+                        default -> {
+                            logger.debug("Command has wrong type, QuantityType or DecimalType required!");
+                            return;
+                        }
+                    }
+                    client.sendCommand(new SetSurplusPowerThresholdCommand(watts));
+                    break;
+                default:
+                    logger.debug("Unknown channel id: {}", channelId);
+            }
+        } catch (IllegalArgumentException e) {
+            Throwable cause = e.getCause();
+            if (cause == null) {
+                logger.debug("Failed to handle command {} for channel {}: {}", command, channelUID, e.getMessage());
+            } else {
+                logger.debug("Failed to handle command {} for channel {}: {} -> {}", command, channelUID,
+                        e.getMessage(), cause.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(FroniusWattpilotConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        FroniusWattpilotConfiguration config = this.config;
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            return;
+        }
+
+        if (config.hostname.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.no-host");
+            return;
+        }
+
+        if (config.password.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.no-password");
+            return;
+        }
+
+        try {
+            client.connect(config.hostname, config.password);
```

Comment:
i guess this starts a http reqauest/response. That should be run from an async thread. The initialize method should be non-blocking and exit fast

Reply:
This starts the Jetty WebSocketClient and calls the asynchronous connect method of it. It does not perform a network requests synchronously.
You can check the implementation at https://github.com/florian-h05/wattpilot4j/blob/89b4ac4c897e94728ae4790d7839bfe565d927fc/src/main/java/com/florianhotze/wattpilot/WattpilotClient.java#L150.